test.xml

```diff
@@ -0,0 +1,234 @@
+        
+<p>
+          Ensures that the names of abstract classes conforming to some pattern and
+          check that <code>abstract</code> modifier exists.
+        </p>
+        
+<p>
+          Rationale: Abstract classes are convenience base class implementations of
+          interfaces. For this reason, it should be made obvious that a given class
+          is abstract by prefacing the class name with 'Abstract'.
+        </p>
+      </subsection>
+
+      <a name="Properties"></a><subsection id="Properties" name="Properties">
+        
+<div class="wrapper">
+          
+<table border="0">
+            
+<tr class="a">
+              
+<th>name</th>
+              
+<th>description</th>
+              
+<th>type</th>
+              
+<th>default value</th>
+              
+<th>since</th>
+            </tr>
+            
+<tr class="b">
+              
+<td>format</td>
+              
+<td>Specify valid identifiers.</td>
+              
+<td><a href="../../property_types.html#Pattern">Pattern</a></td>
+              
+<td><code>&quot;^Abstract.+$&quot;</code></td>
+              
+<td>3.2</td>
+            </tr>
+            
+<tr class="a">
+              
+<td>ignoreModifier</td>
+              
+<td>
+                Control whether to ignore checking for the
+                <code>abstract</code> modifier on classes that match the
+                name.
+              </td>
+              
+<td><a href="../../property_types.html#boolean">boolean</a></td>
+              
+<td><code>false</code></td>
+              
+<td>5.3</td>
+            </tr>
+            
+<tr class="b">
+              
+<td>ignoreName</td>
+              
+<td>
+                Control whether to ignore checking the name. Realistically
+                only useful if using the check to identify that match name
+                and do not have the <code>abstract</code> modifier.
+              </td>
+              
+<td><a href="../../property_types.html#boolean">boolean</a></td>
+              
+<td><code>false</code></td>
+              
+<td>5.3</td>
+            </tr>
+          </table>
+        </div>
+      </subsection>
+
+      <a name="Examples"></a><subsection id="Examples" name="Examples">
+        
+<p>
+          To configure the check:
+        </p>
+        
+          
+          
+        <source>
+&lt;module name=&quot;Checker&quot;&gt;
+  &lt;module name=&quot;TreeWalker&quot;&gt;
+    &lt;module name=&quot;AbstractClassName&quot;/&gt;
+  &lt;/module&gt;
+&lt;/module&gt;
+
+
+</source>
+        
+<p>Example:</p>
+        
+          
+          
+        <source>
+  abstract class AbstractFirst {}
+  abstract class Second {} // violation 'must match pattern'
+  class AbstractThird {} // violation 'must be declared as 'abstract''
+  class Fourth {}
+  abstract class GeneratorFifth {}
+  // violation above 'must match pattern'
+  class GeneratorSixth {}
+</source>
```

Comment:
Macro in action.

The problem is that the output loses it's format :/. I will check it out.

Reply:
https://github.com/checkstyle/checkstyle/blob/0169b0891563cd7e12d39aa42acc6e6b2cb0ee38/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abstractclassname/Example1.java#L14-L22
Formatting is the same as the input file. I am not seeing anything lost.

Reply:
I am talking about the xdoc output. The file itself is all over the place.

Reply:
To me this makes sense. It probably just generates the output as it goes, and doesn't care about spacing or provide context to care. You can see the same in the generated HTML that doesn't involve our macro. We really shouldn't care about this output as it is a generated file, and out of our control.

````
<p>Since Checkstyle 3.2</p>
      <a name="Description"></a><section id="Description">
<h3><a name="Description"></a>Description</h3>
        
<p>
          Ensures that the names of abstract classes conforming to some pattern and
          check that <code>abstract</code> modifier exists.
        </p>
        
<p>
          Rationale: Abstract classes are convenience base class implementations of
          interfaces. For this reason, it should be made obvious that a given class
          is abstract by prefacing the class name with 'Abstract'.
        </p>
      </section>

      <a name="Properties"></a><section id="Properties">
<h3><a name="Properties"></a>Properties</h3>
````

Reply:
Alright, thanks.